### PR TITLE
[RW-10825][risk=low] Support export Terra Warehouse table into AoU reporting for app usage tracking

### PR DESCRIPTION
**Pull request by yonghaoy**
Screenshot 2023-10-18 at 12 35 11 PM

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java [hunk 1]
```diff
@@ -582,6 +594,46 @@ public int getWorkspaceCount() {
...
+    final QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(
+                "SELECT appId, appName, status, createdDate, destroyedDate, "
+                    + "startTime, stopTime, creator, customEnvironmentVariables "
+                    + "FROM `"
+                    + workbenchConfigProvider.get().reporting.terraWarehouseLeoAppUsageTableId
+                    + "` au "
+                    + "JOIN `"
+                    + workbenchConfigProvider.get().reporting.terraWarehouseLeoAppUsageTableId
+                    + "` a on a.id = au.appId where STARTS_WITH(appName, \"all-of-us-\")")
```
Comment (Qi77Qi) [inline api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java:612]: is this `all-of-us` string defined somewhere else as well? Thinking how we can prevent breakage down the line if we decide to change naming convention for some reason down the line

File: api/config/config_preprod.json [hunk 2]
```diff
@@ -145,7 +145,8 @@
   },
   "reporting": {
     "dataset": "reporting_preprod",
-    "maxRowsPerInsert": 500
+    "maxRowsPerInsert": 500,
+    "exportTerraDataWarehouse": false
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply api/config/config_preprod.json:149]: `preprod` uses Terra prod (and also AoU `staging`, `stable`). So we just need reporting in prod env.

File: api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java (hunk 1 above)
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java:612]: ah, good point, I will use this constant: https://github.com/all-of-us/workbench/blob/d2faf20e13a24c7627b5f148ec8fe15e62b9f4f8/api/src/main/java/org/pmiops/workbench/db/model/DbUser.java#L47

Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java:612]: I don't think we want to change this as app name is also GKE service name, and that is how we do egress alert

File: api/config/config_preprod.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply api/config/config_preprod.json:149]: you mean AoU's preprod, staging, stable usage will all be covered by the `prod` data warehouse?

**Review by Qi77Qi — APPROVED** [review]

File: api/config/config_preprod.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (yonghaoy) [reply api/config/config_preprod.json:149]: yup!

**Pull request merged**